### PR TITLE
feat(jsx-directive): add volar plugin

### PR DESCRIPTION
**Pull request by zhiyuanzmj**
Add `v-for` volar plugin.
image

### CODE REVIEW

File: packages/jsx-directive/src/core/index.ts [hunk 1]
```diff
@@ -15,7 +16,10 @@ export function transformJsxVueDirective(code: string, id: string) {
     ast: Program
     offset: number
   }[] = []
-  if (lang === 'vue') {
+  if (
+    lang === 'vue' ||
+    (REGEX_SETUP_SFC.test(id) && code.startsWith('<script '))
```
Comment (sxzz) [inline packages/jsx-directive/src/core/index.ts:21]: Why check `&& code.startsWith('<script ')` here?

File: packages/volar/src/jsx-directive.ts [hunk 2]
```diff
@@ -0,0 +1,122 @@
+import { FileKind, FileRangeCapabilities } from '@volar/language-core'
+import {
+  type Segment,
+  type Sfc,
+  type VueLanguagePlugin,
+  replaceSourceRange,
+} from '@vue/language-core'
+
+function transformDefineOptions({
```
Comment (sxzz) [inline packages/volar/src/jsx-directive.ts:9]:
```suggestion
function transformJsxVueDirective({
```

File: packages/jsx-directive/src/core/index.ts (hunk 1 above)
Comment (zhiyuanzmj) [reply packages/jsx-directive/src/core/index.ts:21]: for test `v-else-if.setup.tsx`

Comment (zhiyuanzmj) [reply packages/jsx-directive/src/core/index.ts:21]: how about change to `v-else-if.tsx` for tests

**Review by sxzz — APPROVED** [review]

**Pull request merged**